source/common/filter/http/filter_config_discovery_impl.cc

```diff
@@ -66,6 +67,21 @@ void DynamicFilterConfigProviderImpl::onConfigUpdate(Envoy::Http::FilterFactoryC
       });
 }
 
+void DynamicFilterConfigProviderImpl::onConfigRemoved(Config::ConfigAppliedCb cb) {
+  tls_.runOnAllThreads(
+      [cb](OptRef<ThreadLocalConfig> tls) {
+        tls->config_ = absl::nullopt;
+        if (cb) {
+          cb();
+        }
+      },
+      [this]() {
+        // This happens after all workers have discarded the previous config so it can be safely
+        // deleted on the main thread by an update with the new config.
+        this->current_config_ = absl::nullopt;
+      });
+}
+
```

Comment:
I am not fully sure but I know some config has default config.
Maybe we should start with defining the exact remove semantic for such config?

CC @kyessenov

Reply:
It should definitely revert back to the default config in that case. I had assumed that we fall back to when the provider returned empty, but it seems like it uses the standard update mechanism to fill in the default config, so seems like this PR needs some work